maanika/sound-map
# Object Detction Feature

## Description

This PR will add the object detection feature to the system. The code consists of sampling distance readings from three ultrasonic rangers and providing a pulse width output to the three motors depending on the distance sampled.

## Screenshots

Hardware set-up
![objdetec](https://user-images.githubusercontent.com/55540980/113497693-bebada80-9549-11eb-97ed-79f984314771.jpg)


## Steps to test

1. Clone the repo and rename project folder appropriately.
2. Build the project using PSoC creator.
3. Ensure the constant "OBJ_DETECT" is set to 1 in "mode.h" file.
4. Observe the motor turning on when the obstacle is less than 150cm and the width of the pulse increases as the distance reduces.
